package.json

```diff
@@ -46,8 +46,9 @@
     "format": "prettier --write '**/*.{ts,tsx,js}'",
     "format-check": "prettier --check '**/*.{ts,tsx,js}'",
     "jest": "jest",
-    "snapshot": "jest --updateSnapshot",
+    "jestUpdate": "jest --updateSnapshot",
     "test": "yarn format-check && yarn tslint && yarn jest",
+    "testUpdate": "yarn format-check && yarn tslint && yarn jestUpdate",
```

Comment:
Not sure if this sequence is so frequently used that it warrants its own script. At least, that's been my experience.

@LaRuaNa WDYT?

Reply:
Well I don't see this command as necessary. Please explain me how does it make sense to `format-check`,  linter check and update snapshots in a single command. I think `update-snapshots` is enough. And`testUpdate` is a copy of `test` anyway. :]

Reply:
in general it would likely be good for changes like this to move to their own PR... it is not obvious from the title of this PR that this was a change that would have landed as part of this

Reply:
So, the problem here is that we have snapshots based testing. Whenever there's a content change in the components like in this PR the tests break. 

So, I end up doing this

```
yarn jestUpdate && yarn test
```

The problem with this is that `jest` runs twice and it's just wasted time in running 15 tests twice for updating the snapshots. So, instead of running tests twice. I added `testUpdate` which updates the things needed before running the jest tests.

Here's the GIF of what I am talking about → https://on.ahmda.ws/751c44

![](https://on.ahmda.ws/751c44/c)

With `testUpdate` the tests run only once since I know there are changes in components so what I do is I run the update before tests.

Reply:
With `testUpdate` the tests run only for once and only people who run this script are sure that they need to update stuff i.e. snapshots before running the tests. 

![](https://on.ahmda.ws/136f01/c)

Reply:
Okay I see your point. It's just not a workflow I'm used to. I consider it as a better workflow to go through failing test cases, understand the problem and update single test case and check the snapshot. I mean what's the benefit of snapshot tests if we just update them all when they fail.

Jest offers such CLI which I quite like :]
<img width="589" alt="image" src="https://user-images.githubusercontent.com/2481234/55575032-32813d00-570e-11e9-97b4-99a9b3bd1bf2.png">

Reply:
@MylesBorins would be careful about that it was an unintentional miss from my end.

Reply:
@LaRuaNa I agree. That's why I made a new script. The `jest` cli in test doesn't wait and watch for changes when it fails during `yarn test` otherwise I also prefer just pressing `u` to update the snapshots. That's why in my personal projects I go with `testUpdate`. 

What are your thoughts on this now. If you agree let's keep it that way otherwise, I'll open a PR to reverse these changes and discuss a more neutral ground on what feels like a better developer experience. 

Peace! ✌️

Reply:
@ahmadawais Just fix please the naming convention for the sake of consistency.

Reply:
Perhaps having a script that runs jest in watch mode is more useful in general to everyone. This way you don't have to keep running this script everytime you edit a file. Also, you now get to use jest's [interactive snapshot mode](https://jestjs.io/docs/en/snapshot-testing#interactive-snapshot-mode).

```diff
- "testUpdate": "yarn format-check && yarn tslint && yarn jestUpdate"
+ "test-watch": "yarn jest --watch",
```

Reply:
Raised a PR with the above change: https://github.com/nodejs/nodejs.dev/pull/230. PTAL.